builtin/providers/docker/data_source_docker_image.go

```diff
@@ -0,0 +1,78 @@
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/heroku/docker-registry-client/registry"
+)
+
+func dataSourceDockerImage() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceDockerImageRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"id": &schema.Schema{
```

Comment:
I believe that `id` is in the schema by default settable via `d.SetId()` and readable via `d.Id()` - I'm actually surprised the internal schema validation isn't erroring out. It should probably be catching such cases.